relentless/ensemble.py

```diff
@@ -8,132 +12,273 @@
         return 1./(self.kB*self.T)
 
     @property
     def V(self):
+        """:py:class:`Volume`: The volume of the system, can only be set as a `Volume` object."""
         return self._V
 
     @V.setter
     def V(self, value):
-        if 'V' in self.conjugates:
-            self._V = value
-        else:
-            raise AttributeError('Volume is not a conjugate variable.')
+        if not isinstance(value, Volume):
+            raise TypeError('V can only be set as a Volume object.')
+        self._V = value
 
     @property
     def P(self):
+        """float or int: The pressure of the system."""
         return self._P
 
     @P.setter
     def P(self, value):
-        if 'P' in self.conjugates:
-            self._P = value
-        else:
-            raise AttributeError('Pressure is not a conjugate variable.')
+        self._P = value
 
     @property
     def N(self):
-        return self._N.todict()
-
-    @N.setter
-    def N(self, value):
-        for t in value:
-            if 'N_{}'.format(t) in self.conjugates:
-                self._N[t] = value[t]
-            else:
-                raise AttributeError('Number is not a conjugate variable.')
+        """:py:class:`FixedKeyDict`: The number of particles for each specified type."""
+        return self._N
 
     @property
     def mu(self):
-        return self._mu.todict()
+        """:py:class:`FixedKeyDict`: The chemical potential for each specified type."""
+        return self._mu
 
-    @mu.setter
-    def mu(self, value):
-        for t in value:
-            if 'mu_{}'.format(t) in self.conjugates:
-                self._mu[t] = value[t]
-            else:
-                raise AttributeError('Chemical potential is not a conjugate variable.')
+    @property
+    def types(self):
+        """tuple: The types in the ensemble."""
+        return tuple(self._N.todict().keys())
 
     @property
-    def conjugates(self):
-        return self._conjugates
+    def constant(self):
+        """dict: The constant variables in the system. READ-ONLY."""
+        return self._constant
 
-    def reset(self):
-        if 'V' in self.conjugates:
+    def clear(self):
+        """Clears all conjugate variables and all rdf values in the ensemble
+        (set to `None`).
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            Returns the ensemble object with cleared parameters.
+
+        """
+        if not self.constant['V']:
             self._V = None
-        if 'P' in self.conjugates:
+        if not self.constant['P']:
             self._P = None
         for t in self.types:
-            if 'N_{}'.format(t) in self.conjugates:
+            if not self.constant['N'][t]:
                 self._N[t] = None
-            if 'mu_{}'.format(t) in self.conjugates:
+            if not self.constant['mu'][t]:
                 self._mu[t] = None
-        self.rdf = core.PairMatrix(self.types)
+        self.rdf = core.PairMatrix(types=self.types)
```

Comment:
Just in case someone is holding a reference to the RDF pair matrix, let's just reset these. I think this should work:
```suggestion
        for pair in self.rdf:
            self.rdf[pair] = None
```